test/integration/scheduler/util.go

```diff
@@ -54,6 +56,60 @@ type TestContext struct {
 	scheduler              *scheduler.Scheduler
 }
 
+// createConfigurator create a configurator for scheduler with given informer factory and default name.
+func CreateConfigurator(
+	clientSet clientset.Interface,
+	informerFactory informers.SharedInformerFactory,
+) scheduler.Configurator {
+	// Enable EnableEquivalenceClassCache for all integration tests.
+	utilfeature.DefaultFeatureGate.Set("EnableEquivalenceClassCache=true")
```

Comment:
modifying global state here without restoring the original value is not something we should do... this also assumes no other tests are running in parallel... is that the case?

Reply:
Ya I need to restore it.
 
According to other tests, it seems we already assumed no other tests are running in parallel. So do the Makefile.

Reply:
@liggitt Oh, what if I do want to enable it for all tests? (because technically there's no test specifically for equiv class in integration)

btw, I just noticed there are some other places did not restore the feature gate ... should I fix them all? e.g. `// Enable TaintNodeByCondition`

Reply:
We have a `SetFeatureGateDuringTest`.

https://github.com/kubernetes/kubernetes/blob/6827c3cf4725e3490dc27aabd5a43a26de3363ef/staging/src/k8s.io/apiserver/pkg/util/feature/testing/feature_gate_testing.go#L26-L40

Reply:
@nikhita Emm, that's helpful, let me update the PR. Thanks!

Reply:
@resouer Please make sure that we use `initTest` in util.go as much as possible in our integration tests. There are cases where we need to configure a special scheduler. Those cannot use `initTest`. They can use these configurators, but those are few cases and I believe eCache is not that important there (although I haven't looked carefully). If my understanding is correct, then you can use `SetFeatureGateDuringTest` in `initTest`.